ddtrace/internal/telemetry/writer.py

```diff
@@ -32,25 +32,67 @@ def _get_interval_or_default():
     return float(os.getenv("DD_TELEMETRY_HEARTBEAT_INTERVAL", default=60))
 
 
+class _TelemetryClient:
+    def __init__(self, endpoint):
+        # type: (str) -> None
+        self._agent_url = get_trace_url()
+        self._endpoint = endpoint
+        self._encoder = JSONEncoderV2()
+        self._headers = {
+            "Content-type": "application/json",
+            "DD-Telemetry-API-Version": "v1",
+        }
+
+    @property
+    def url(self):
+        return "%s/%s" % (self._agent_url, self._endpoint)
```

Comment:
Why is `agent_url` fixed to always use `get_trace_url()` now?

Reply:
The `agent_url` argument in Telemetry.__init__(.....) was never used, so we always defaulted to using get_trace_url() anyway. The telemetry only submits payloads to the datadog agent. We can configure this value using the DD_TRACE_AGENT_URL configuration.  I'll update the PR description to include this